src/containerbuddy/config.go

```diff
@@ -12,7 +12,7 @@ import (
 )
 
 type Config struct {
-	DiscoveryUri string           `json:"consul"` // TODO: allow choice of discovery here
+	Consul	     *string          `json:"consul"`
```

Comment:
We should mark this field as `omitempty` so that it's not serialized if we want to do that in the future. I'd also prefer we keep this as a string rather than a pointer and do a check for `""` rather than `nil` in the switch block on line 76. What you have here works but avoiding a scenario where we pass around `nil` will reduce the risk of future bugs.